ddtrace/tracer/payload.go

```diff
@@ -74,9 +77,8 @@ func (p *payload) size() int {
 	return p.buf.Len() + len(p.header) - p.off
 }
 
-// reset should *not* be used. It is not implemented and is only here to serve
-// as information on how to implement it in case the same payload object ever
-// needs to be reused.
+// reset sets up the payload to be read a second time. It maintains the
+// underlying byte contents of the buffer.
```

Comment:
Let's actually *not* delete this comment. The original idea behind `reset` was to reuse `payload` for another set of traces after a successful delivery by resetting the buffer to empty. This (in principle) would save memory since the same backing byte buffer could be used, avoiding GC churn.

This note is here to remind everyone why we don't do this (after having tried). I think this should be updated and moved to the `payload` type documentation, along with the warning inside the function since `reset` is no longer a function that prepares the payload for reuse. Basically, a note on the type about why we don't/can't reuse `payload` objects for multiple payloads.